### ISSUE TITLE
Do not flag missing nonce check inside conditional on pure comparison

### ISSUE THREAD

**Issue by GaryJones** [WordPress-Coding-Standards/WordPress-Coding-Standards#1506]
Original issue: https://github.com/Automattic/VIP-Coding-Standards/issues/39

```
<?php

$a = function() {
    if (
		'newsletter_sign_up' === $_POST['action'] &&
		wp_verify_nonce( $_POST['tc_event_info_newsletter_nonce'] )
	) {
    }
}
```

Produces following errors:

```
---------------------------------------------------------------------------------------------
FOUND 2 ERRORS AND 2 WARNINGS AFFECTING 2 LINES
---------------------------------------------------------------------------------------------
 5 | WARNING | Detected access of super global var $_POST, probably needs manual inspection.
 5 | ERROR   | Processing form data without nonce verification.
 6 | WARNING | Detected access of super global var $_POST, probably needs manual inspection.
 6 | ERROR   | Detected usage of a non-validated nor non-sanitized input variable: $_POST
---------------------------------------------------------------------------------------------
```

Simply swapping the conditionals fixes the issue:

```
<?php

$a = function() {
    if (
		wp_verify_nonce( $_POST['tc_event_info_newsletter_nonce'] ) &&
		'newsletter_sign_up' === $_POST['action']
	) {
    }
}
```

```
---------------------------------------------------------------------------------------------
FOUND 1 ERROR AND 2 WARNINGS AFFECTING 2 LINES
---------------------------------------------------------------------------------------------
 5 | WARNING | Detected access of super global var $_POST, probably needs manual inspection.
 5 | ERROR   | Detected usage of a non-validated nor non-sanitized input variable: $_POST
 6 | WARNING | Detected access of super global var $_POST, probably needs manual inspection.
---------------------------------------------------------------------------------------------
```

Related code (starting points):

* https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/aa75d75f6fd5720b25330abb243a94197538e127/WordPress/Sniffs/CSRF/NonceVerificationSniff.php#L78
* https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/aa75d75f6fd5720b25330abb243a94197538e127/WordPress/Sniff.php#L852

**Comment by jrfnl** [comment]
Loosely related to #1497
